source/appModules/soffice.py

```diff
@@ -208,11 +209,24 @@ def announceSelectionChange(self):
 
 	def _get_cellCoordsText(self):
 		if self.hasSelection and controlTypes.State.FOCUSED in self.states:
-			selected, count = self.table.IAccessibleTable2Object.selectedCells
-			firstAccessible = selected[0].QueryInterface(IA2.IAccessible2)
+			count = self.table.IAccessibleTable2Object.nSelectedCells
+			selection = self.table.IAccessibleObject.accSelection
+			enumObj = selection.QueryInterface(oleacc.IEnumVARIANT)
+			firstChild, _retrievedCount = enumObj.Next(1)
+			# skip over all except the last element
+			enumObj.Skip(count - 2)
+			lastChild, _retrieveCount = enumObj.Next(1)
+			# in LibreOffice 7.3.0, the IEnumVARIANT returns a child ID,
+			# in LibreOffice >= 7.4, it returns an IDispatch
+			if isinstance(firstChild, int):
+				tableAccessible = self.table.IAccessibleTable2Object.QueryInterface(IA2.IAccessible2)
+				firstAccessible = tableAccessible.accChild(firstChild).QueryInterface(IA2.IAccessible2)
+				lastAccessible = tableAccessible.accChild(lastChild).QueryInterface(IA2.IAccessible2)
+			else:

```

Comment:
I think it would be best to capture the type specifically here, and raise an exception if an unexpected type is used.

```suggestion
			elif isinstance(firstChild, IDispatch):  # from comtypes.automation import IDispatch
				...
			else:
				raise RuntimeError(f"Unexpected LibreOffice object {firstChild}, type: {type(firstChild)}")
```

Reply:
That sounds reasonable. I have added a commit on top doing so. The exact type used here is comtypes.client.dynamic._Dispatch, not comtypes.automation.IDispatch, so that commit uses this.